### ISSUE TITLE
Conversion: timestamp conversions

### ISSUE THREAD

**Issue by fubuloubu** [ApeWorX/ape#404]
### Overview
Convert stringified time values to timestamp/timedelta integer values

### Specification
```py
c = dev.deploy(project.VestingContract, "2022-01-30T00:00:00.000Z")  # Convert ISO timestamps
contract.claim("1 hour")  # Convert timedelta in call args
assert contract.waitPeriod() == "1 week"  # Use it in return types for comparisions?
```

**Comment by johnson2427** [comment]
datetime has a string conversion package.
Do you want it to specify for UTC timezone?
Is that the only format of date and time you are looking for?
```python
from datetime import datetime, timedelta

datetime_obj = datetime.fromisoformat("2022-01-30T00:00:00.000Z"[:-1])
datetime_obj = datetime.strptime(datetime_obj, "%Y-%m-%d %H:%M:%S")
change_time = timedelta(hours=1)
new_datetime_obj = datetime_obj + change_time
```
This will give you a one hour delta. If you want 1 week, use `timedelta(weeks=1)`

**Comment by fubuloubu** [comment]
> Do you want it to specify for UTC timezone?

I think to be un-ambigious, it has to be

> Is that the only format of date and time you are looking for?

alternative timezones are probably okay, but should be converted to UTC first before creating the timestamp from parsing the string

**Comment by johnson2427** [comment]
I just didn't know if all the data we are grabbing runs through UTC, if figured it did.
But are there other formats? Like would you see Y-d-m, like you do in Europe? or Jan 30, 2022 00:00:00.00000000? If so, this causes some regular expressions and some headache trying to convert down
Just curious if this becomes a complicated problem, or not. But that code will handle the timedelta

**Comment by fubuloubu** [comment]
If you see an amigious timestamp, you can assume the same timezone as local, but that would probably cause some compatibility errors for users in different locales

**Comment by johnson2427** [comment]
We can have it pull current data, and check time against UTC, if it's equal, then keep UTC, if not, search for the timezone the data was pulled for

**Comment by fubuloubu** [comment]
> We can have it pull current data, and check time against UTC, if it's equal, then keep UTC, if not, search for the timezone the data was pulled for

if I write a script that says "12:00" and upload it, and then someone 5000 miles away pulls it down... what's the timestamp?

**Comment by johnson2427** [comment]
If you are making a script with a time for a future date, force them to add timezone, then we convert that timezone to UTC

**Comment by fubuloubu** [comment]
> If you are making a script with a time for a future date, force them to add timezone, then we convert that timezone to UTC

Probably best not to include any feature like this that implicitly finds the timezone based on local context, at least until we can figure out how to do it right in a future PR. I'd start with only allowing UTC first.

**Comment by johnson2427** [comment]
This was completed. I am not adding a convert datetime object

**Comment by fubuloubu** [comment]
p.s. next time if you add `fix: #404` to your PR, it will automatically close!

**Comment by fubuloubu** [comment]
closed by: https://github.com/ApeWorX/ape/pull/432
